### PR TITLE
Travis pysat requirements

### PR DESCRIPTION
**Pull request by aburrell**
Addresses issue #30 and uses new pysat style standard for setup.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,17 +32,19 @@ install:
...
   # Create conda test environment
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION numpy scipy pandas xarray requests beautifulsoup4 lxml netCDF4 h5py pytest-cov pytest-ordering coveralls future
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION
   - conda activate test-environment
-  # Dependencies not available through conda, install through pip
-  - pip install pytest-flake8
+  # Once pysat 3.0 is released, remove the custom pysat install and change
+  # this to:
+  # - while read requirement; do conda install --yes $requirement; done < requirements.txt
+  - conda install --yes h5py numpy pandas xarray
```
Comment (jklenzing) [inline .travis.yml:40]:
```suggestion
  - conda install --yes h5py numpy pandas xarray portalocker
```

File: setup.cfg [hunk 2]
```diff
@@ -1,3 +1,50 @@
+[metadata]
+name = pysatMadrigal
+version = file: pysatMadrigal/version.txt
+url = https://github.com/pysat/pysatMadrigal
+author = Angeline G. Burrell, et al.
+author_email = angeline.burrell@nrl.navy.mil
```
Comment (jklenzing) [inline setup.cfg:6]: Should we be using individual emails here or the developer's email?

**Review by jklenzing — COMMENTED** [review]
Our current install method on Travis is not auto-installing `portalocker`, which causes the pysat install to fail.  This could be because we are using a custom install through git rather than pip.  Options:
- hardwire an install for now (suggested code below)
- add to requirements

File: setup.cfg (hunk 2 above)
Comment (aburrell) [reply setup.cfg:6]: I think the pysat development email would be better, but it currently isn't standard.  Happy to update it here.

File: .travis.yml (hunk 1 above)
Comment (aburrell) [reply .travis.yml:40]: This isn't needed because a few lines down we go and get that from the pysat requirements file.

File: .travis.yml [hunk 3]
```diff
@@ -28,17 +32,19 @@ install:
...
   - pip install madrigalWeb
-  - pip install PyForecastTools
-  - pip install pysatCDF >/dev/null
   # Custom pysat install
   - cd ..
   - git clone --single-branch --branch develop-3 https://github.com/pysat/pysat.git
   - cd pysat
+  # Install the pysat requirements
+  - while read requirement; do conda install --yes $requirement; done < requirements.txt
```
Comment (aburrell) [inline .travis.yml:47]: @jklenzing this makes sure all pysat requirements are installed until we go to installing pysat through pip.

File: setup.cfg (hunk 2 above)
Comment (aburrell) [reply setup.cfg]: Opps, this is a new standard I made, isn't it? :P

**Review by jklenzing — APPROVED** [review]
Looks good!  The merge conflict is probably related to my quick fix from the last PR.  The changes here should supersede that.

**Pull request merged**